Bring in loopback4-example-rpc-server

bajtos opened the pull request:
The first commit 929737c brings in an exact copy of https://github.com/strongloop/loopback4-example-rpc-server.

The second commit f8d08a5 contains changes necessary to integrate the example with the rest of the monorepo.

See #836

<!--
Please provide a high-level description of the changes made by your pull request.

Include references to all related GitHub issues and other pull requests, for example:

Fixes #123
Implements #254
See also #23
-->


## Checklist

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] Related API Documentation was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [x] Affected example projects in `packages/example-*` were updated

b-admike commented on packages/example-rpc-server/README.md:
````diff
@@ -1,14 +1,29 @@
+npm i -g @loopback/cli
+```
+
+2. Download the "getting-started" application.
````
I'm guessing you're adding this to the `lb4 example` prompt and editing these instructions?

virkt25 commented on packages/example-rpc-server/package.json:
```diff
@@ -0,0 +1,62 @@
+    "loopback-application",
+    "loopback"
+  ],
+  "main": "./dist/src/index.js",
```
Shouldn't this point to `index.js`?

virkt25 replied:
Actually it appears that `index.d.ts` and `index.js` are missing in this example yet it supports Node 6 so I'm actually confused as to how this even passes on 6 rn.

virkt25 commented on packages/example-rpc-server/test/mocha.opts:
```diff
@@ -0,0 +1 @@
+--recursive
```
This file can be removed as it's not used by the test script in `package.json`

virkt25 commented on packages/example-rpc-server/tslint.build.json:
```diff
@@ -0,0 +1,17 @@
+{
```
We don't need to a tslint file when using `lb-tslint`. This file can be removed.

virkt25 commented on packages/example-rpc-server/tslint.json:
```diff
@@ -0,0 +1,4 @@
+{
```
This file can be removed. None of the other `example-*` repos have it.

bajtos commented:
Thank you @kjdelisle and @virkt25 for your detailed review. So many trivial mistakes, I feel bad about not being more diligent when checking my changes myself.

I believe I have addressed all your comments now.

bajtos commented:
FWIW, the build is failing because of a know problem in one of our dependencies, see https://github.com/sinonjs/nise/issues/27 and https://github.com/sinonjs/nise/pull/29
